File: clients/typescript/src/PrintNannyCloudAuth.ts

export class PrintNannyCloudAuth {
  private _email: string;
  private _apiToken: string;
  private _apiUrl: string;

  constructor(input: {
    email: string,
    apiToken: string,
    apiUrl: string,
  }) {
    this._email = input.email;
    this._apiToken = input.apiToken;
    this._apiUrl = input.apiUrl;
  }

  get email(): string { return this._email; }
  set email(email: string) { this._email = email; }

  get apiToken(): string { return this._apiToken; }
  set apiToken(apiToken: string) { this._apiToken = apiToken; }

  get apiUrl(): string { return this._apiUrl; }
  set apiUrl(apiUrl: string) { this._apiUrl = apiUrl; }
}

File: clients/rust/src/print_nanny_cloud_auth.rs
// PrintNannyCloudAuth represents a PrintNannyCloudAuth model.
#[derive(Clone, Debug, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
pub struct PrintNannyCloudAuth {
    #[serde(rename="email")]
    email: String,
    #[serde(rename="api_token")]
    api_token: String,
    #[serde(rename="api_url")]
    api_url: String,
}

impl PrintNannyCloudAuth {
    pub fn new(email: String, api_token: String, api_url: String) -> PrintNannyCloudAuth {
        PrintNannyCloudAuth {
        email,
        api_token,
        api_url,
        }
    }
}
